huxley/www/js/components/AdvisorAssignmentsView.js

```diff
@@ -230,51 +236,24 @@ var AdvisorAssignmentsView = React.createClass({
     var confirm = window.confirm("By pressing okay you are committing to the financial responsibility of each assignment. Are you sure you want to finalize assignments?");
     var school = CurrentUserStore.getCurrentUser().school;
     if (confirm) {
-      this.setState({loading: true});
-      ServerAPI.updateSchool(school.id, {assignments_finalized: true})
-        .then(this._handleFinalizedSuccess, this._handleError);
+      CurrentUserActions.updateSchool(school.id, {
+        assignments_finalized: true,
+      });
     }
   },
 
   _handleAssignmentDelete: function(assignment) {
     var confirm = window.confirm("Are you sure you want to delete this assignment?");
     if (confirm) {
-      this.setState({loading: true});
-      ServerAPI.updateAssignment(assignment.id, {rejected: true}).then(
-        this._handleAssignmentDeleteSuccess.bind(this, assignment.id),
-        this.handleError
-      );
+      AssignmentActions.updateAssignment(assignment.id, {
+        rejected: true,
+      });
     }
   },
 
   _handleSave: function(event) {
     var school = CurrentUserStore.getCurrentUser().school;
     DelegateActions.updateDelegates(school.id, this.state.delegates);
-  },
-
-  _handleFinalizedSuccess: function(response) {
-    CurrentUserActions.updateSchool(response);
-    this.setState({loading: false});
```

Comment:
This is definitely for another PR (and not necessarily hi-pri), but just putting it on your radar:

Perhaps there should be some kind of global "loading" state. Since updates are optimistic, it's possible that a user could make a change and then close the browser before the change is reflected in the server (possibly due to poor network conditions or something). In that case, we want to show a dialog prompting the user to wait a little longer while the request completes.